### PR TITLE
local-gadget: Update documentation

### PR DESCRIPTION
**Pull request by blanquicet**
# Update local-gadget documentation

This PR updates the [local-gadge documentation](https://github.com/kinvolk/inspektor-gadget/blob/jose/local-gadget-docs/docs/local-gadget.md).

This PR goes on top of https://github.com/kinvolk/inspektor-gadget/pull/763.

### CODE REVIEW

File: cmd/kubectl-gadget/profile/cpu.go [hunk 1]
```diff
@@ -148,7 +145,7 @@ func (p *CPUParser) DisplayResultsCallback(traceOutputMode string, results []str
 			}
 		}
 
-		p.PrintColumnsHeader()
+		p.BuildColumnsHeader()
```
Comment (mauriciovasquezbernal) [inline cmd/kubectl-gadget/profile/cpu.go:148]: It's missing the logic to print the returned value.

File: docs/local-gadget.md [hunk 2]
```diff
@@ -5,41 +5,203 @@ description: >
   How to use the local-gadget tool.
 ---
 
-Inspektor Gadget can also be used without Kubernetes to trace containers with the `local-gadget` tool.
+Inspektor Gadget can also be used without Kubernetes to trace containers with
+the `local-gadget` tool.
+
+Currently, the `local-gadget` can trace containers managed by Docker regardless
+of whether they were created via Kubernetes or not. In addition, it will also
+use the CRI to trace containers managed by Containerd and CRI-O, meaning only
```
Comment (mauriciovasquezbernal) [inline docs/local-gadget.md:13]: `Containerd` -> `containerd` (as written in https://containerd.io/).

**Review by mauriciovasquezbernal — COMMENTED** [review]
> Therefore, I decided to [merge them](https://github.com/kinvolk/inspektor-gadget/blob/271ec370854b479a1265bdb6e198fa381da4488a/cmd/common/utils/parser.go#L23-L36) and add a SeparateWithTaps which allows specifying whether the parser has to print using taps or the requested width. If preferred, I can make a separate PR for it.

I think that'd help a lot with the review. Otherwise this PR contains very unrelated changes.

**Comment by blanquicet** [comment]
> If preferred, I can make a separate PR for it.
> > I think that'd help a lot with the review. Otherwise this PR contains very unrelated changes.
> >
@mauriciovasquezbernal You are right. I divided this PR into three steps to make the review easier:

1. https://github.com/kinvolk/inspektor-gadget/pull/762
2. https://github.com/kinvolk/inspektor-gadget/pull/763
3. https://github.com/kinvolk/inspektor-gadget/pull/760

PTAL.

File: cmd/common/utils/parser.go [hunk 3]
```diff
@@ -19,24 +19,55 @@ import (
...
+// BaseParser is a base for a parser that implements the most common methods.
 type BaseParser struct {
+	// ColumnsWidth are the columns that can be configured to be printed with
+	// the width to be used.
 	ColumnsWidth map[string]int
+
+	// SeparateWithTaps defines whether the columns has to be separated with the
+	// width defined in ColumnsWidth (if false), or with taps ignoring the width
+	// defined in ColumnsWidth (if true).
+	SeparateWithTaps bool
```
[the remark this answers is not in the document]
Comment (blanquicet) [reply cmd/common/utils/parser.go]: Managed in https://github.com/kinvolk/inspektor-gadget/pull/762.

File: cmd/kubectl-gadget/profile/cpu.go (hunk 1 above)
Comment (blanquicet) [reply cmd/kubectl-gadget/profile/cpu.go]: Managed in https://github.com/kinvolk/inspektor-gadget/pull/762

File: docs/local-gadget.md (hunk 2 above)
Comment (blanquicet) [reply docs/local-gadget.md]: Applied everywhere, thanks!

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM.

**Pull request merged**